platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/hashing/DefaultEventHasher.java

```diff
@@ -16,16 +16,39 @@
 
 package com.swirlds.platform.event.hashing;
 
+import com.hedera.hapi.node.base.SemanticVersion;
 import com.swirlds.platform.event.PlatformEvent;
+import com.swirlds.state.spi.HapiUtils;
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * Default implementation of the {@link EventHasher}.
  */
 public class DefaultEventHasher implements EventHasher {
+    private final SemanticVersion currentSoftwareVersion;
+    private final boolean migrateEventHashing;
+
+    /**
+     * Constructs a new {@link DefaultEventHasher} with the given {@link SemanticVersion} and migration flag.
+     *
+     * @param currentSoftwareVersion the current software version
+     * @param migrateEventHashing    if true then use the new event hashing algorithm for new events, events created by
+     *                               previous software versions will still need to be hashed using the old algorithm.
+     */
+    public DefaultEventHasher(final SemanticVersion currentSoftwareVersion, final boolean migrateEventHashing) {
```

Comment:
null annotation (and maybe null checks) missing

Reply:
added